x/chainlink/keeper/msg_server.go

```diff
@@ -1,17 +1,22 @@
 package keeper
 
 import (
+	"context"
+
 	"github.com/ChainSafe/chainlink-cosmos/x/chainlink/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-type msgServer struct {
-	Keeper
-}
+var _ types.MsgServer = Keeper{}
 
-// NewMsgServerImpl returns an implementation of the MsgServer interface
-// for the provided Keeper.
-func NewMsgServerImpl(keeper Keeper) types.MsgServer {
-	return &msgServer{Keeper: keeper}
-}
+// SubmitFeedData implements the tx/SubmitFeedData gRPC method
+func (k Keeper) SubmitFeedData(c context.Context, msg *types.MsgFeedData) (*types.MsgFeedDataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	k.SetFeedData(ctx, msg)
 
-var _ types.MsgServer = msgServer{}
+	// TODO: how to return txHash and height here?
```

Comment:
look at the comments i made above about SetFeedData
https://github.com/ChainSafe/chainlink-cosmos/pull/27/files#diff-aa217046922f981f1638f1b9b95917f21a0a7064634a7387dc9db90a7e1a12e3R42